cosmos/runtime/baseapp/baseapp.go

```diff
@@ -141,6 +145,28 @@ type PolarisBaseApp struct {
 	ERC20Keeper *erc20keeper.Keeper
 }
 
+// InitChainer initializes the chain.
+func (a *PolarisBaseApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain) (abci.ResponseInitChain, error) {
+	// Validate that ethereum genesis and cosmos genesis timestamp are the same.
+	// TODO: investigate how we can make this work in evm/keeper/genesis.go
+	// currently doesn't work in there since the scope of gen is restricted to the evm component
+	if req.Time != ctx.BlockHeader().Time {
```

Comment:
checking against the abci response isn't useful, you want to check against the ethereum genesis.